### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by huzaifakhan0308**
- add form to my blogsVault project

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(
+      text: params[:text],
+      post_id: params[:post_id]
+    )
+    @comment.user_id = current_user.id
```
Comment (MahmoudBakr23) [inline app/controllers/comments_controller.rb]: **OPTIONAL**
- Well done creating the `create` method and the `@comment` instance variable correctly. However, using the `private` keyword in the controllers to create the `params` method is very useful for the application's security and performance. Please learn more [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md).
```ruby
def create
  @comment = current_user.comments.new(comment_params)
  @comment.post_id =  params[:post_id]
  ...
end
private
def comment_params
  params.require(:comment).permit(:text)
end
```

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      post_id: params[:post_id]
+    )
+    if @like.save
+      flash[:notice] = 'Like was successfully created.'
+      redirect_to user_post_path(params[:user_id], params[:post_id])
+    else
+      flash.now[:alert] = 'Like was not created.'
+      render 'posts/show'
+    end
+  end
```
Comment (MahmoudBakr23) [inline app/controllers/likes_controller.rb]: - Please see the optional request about the `params` private method. 🙏

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -8,4 +8,28 @@ def show
...
+    @post.commentscounter = 0
+    @post.likescounter = 0
+    if @post.save
+      flash[:notice] = 'Post was successfully created.'
+      redirect_to user_post_path(params[:user_id], @post)
+    else
+      flash.now[:alert] = 'Post was not created.'
+      render :new
+    end
+  end
```
Comment (MahmoudBakr23) [inline app/controllers/posts_controller.rb]: - Please see the optional request about the `params` private method. 🙏

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -8,4 +8,28 @@ def show
...
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(
+      title: params[:post][:title],
+      text: params[:post][:text]
+    )
+    @post.user_id = params[:user_id]
```
Comment (MahmoudBakr23) [inline app/controllers/posts_controller.rb]: - [ ] Please use the `current_user` method here to identify the post's owner.
`@post = current_user.posts.new(title: params[:title],
      text: params[:text])`
OR
`@post = current_user.posts.new(post_params)` if you used the `params` method as requested as optional.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @huzaifakhan0308  😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done creating the `current_method`
- Good work so far in the controllers' `create` and `new` methods
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
_Please check the comments under the review._

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @huzaifakhan0308, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitfow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**